src/components/SourceMaterialForm.svelte

```diff
@@ -0,0 +1,112 @@
+<script>
+
+    import { createEventDispatcher } from 'svelte'
+
+    let sourceType;
+    let source;
+    let description;
+    
+    let dispatch = createEventDispatcher();
+
+    const validateSubmission = () => {
+        let valid = true;
+        let errorMessage = "ERROR! The following fields are required: "
+        if (sourceType == undefined)
+        {
+            valid = false;
+            errorMessage += "\n-Source type (dropdown menu)"
+        }
+        if (source == "" || source == undefined)
+        {
+            valid = false;
+            errorMessage += "\n-Brief source description or file upload"
+        }
+        if (description == "" || description == undefined)
+        {
+            valid = false;
+            errorMessage += "\n-Additional Description"
+        }
+        if (!valid)
+        {
+            alert(errorMessage);
+        }
+        return valid;;
+    };
+
+    const handleSubmit = () => {
+        if (validateSubmission())
+        {
+        console.log("Submission successful");
+        const sourceMaterial = {
+            id: "sourceMaterial" + Date.now() + "x" + Math.random(),
+            type: "sourceMaterial",
+            sourceType,
+            source,
+            description
+        }
+        dispatch('addSourceMaterial', sourceMaterial)
+    }
+    };
+
+</script>
+
+<div class = "inside">
+    <h1 class = "formType">New Source Material Or Citation</h1>
+    <h2>1. Enter Information</h2>
+    <p><span class = "red">*</span>Required fields</p>
+<form>
+    <p>Source<span class = "red">*</span></p>
+    <select bind:value = {sourceType}>
+        <option value = "undefined"></option>
+        <option value = "Experience">Personal Experience</option>
+        <option value = "Site">Exists in an archived site</option>
+        <option value = "Doc">Exists in published document</option>
+    </select>
+    {#if sourceType == "Experience"}
+    <input class = "larger" type = "text" placeholder = "Briefly describe experience" bind:value = {source} required>
+    {:else if sourceType == "Site"}
+    <input class = "larger" type = "text" placeholder = "Paste url or briefly describe source" bind:value = {source} required>
+    {:else if sourceType == "Doc"}
+    <label>
+        <br/>
+        Upload file here (PDF accepted)
+        <input type = "file" accept = "pdf" bind:value = {source}/>
+    </label>
+    {/if}

```

Comment:
Not sure about the ethics (or storage implications) of vacuuming up documents in a centralized location. An academic citation is the main thing I'd be looking for instead: for a textual source, something like source creator, name of source, name of source container, location of source container, date created, date accessed. For a personal source, name and date is sufficient.